### PR TITLE
Kiuh refactoring

### PR DESCRIPTION
**Pull request by Kiuh**
UIController folder can be removed now

### CODE REVIEW

File: JamGame/Assets/Scripts/UI/UIController/AnimatorsSwitcher.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+[Serializable]
+public class AnematorWithBool
```
Comment (mertwole) [inline JamGame/Assets/Scripts/UI/UIController/AnimatorsSwitcher.cs:6]: Animator

File: JamGame/Assets/Scripts/Level/Sheduler.cs [hunk 2]
```diff
@@ -15,6 +15,20 @@ public class Sheduler : MonoBehaviour
 
         private IEnumerator<DayConfig> dayEnumerator;
         private IEnumerator<IDayAction> actionEnumerator;
+        private bool isPlanned = true;
+
+        // Must be catched by Executor
+        public void ActionEndRealese()
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Level/Sheduler.cs:21]: Sorry what?)

File: JamGame/Assets/Scripts/Level/Finances/Model.cs [hunk 3]
```diff
@@ -10,6 +10,7 @@ public class Model : MonoBehaviour
         [SerializeField, InspectorReadOnly]
         private int money;
         public UnityEvent<int> MoneyChange;
+        public TarrifsCounter TarrifsCounter;
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Level/Finances/Model.cs:13]: As I see it doesn't used anywhere

File: JamGame/Assets/Scripts/TileBuilder/Command.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@ namespace TileBuilder.Command
 {
     public interface ICommand
     {
-        public Result Execute(TileBuilderImpl tile_builder);
+        public Result Execute(TileBuilderImpl tile_builder, ref TileUnionImpl selected_tile);
```
Comment (mertwole) [inline JamGame/Assets/Scripts/TileBuilder/Command.cs:12]: selected_tile don't used in all commands, so ICommand shouldn't know about it

**Review by mertwole — CHANGES_REQUESTED** [review]

File: JamGame/Assets/Scripts/Level/Executor.cs [hunk 5]
```diff
@@ -23,96 +18,86 @@ public class Executor : MonoBehaviour
...
 
-        public void Execute(Working working, Action next_action)
+        public void Execute(Cutscene cutscene)
         {
-            transitionPanel.SetText("Working start.");
-            uiController.PlayDayActionStart(
-                working.GetType(),
-                () => _ = StartCoroutine(WorkingTime(working.WorkingTime, next_action))
-            );
+            transitionPanel.SetText(cutscene.Text);
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Level/Executor.cs:92]: transitionPanel is View and Executor is Model so remove this dependency

**Review by mertwole — CHANGES_REQUESTED** [review]

File: JamGame/Assets/Scripts/AnimatorsSwitcher/Model.cs
Comment (mertwole) [inline JamGame/Assets/Scripts/AnimatorsSwitcher/Model.cs:1]: Rename this file to AnimatorsSwitcher.cs

File: JamGame/Assets/Scripts/Common/CenterOfMass.cs [hunk 6]
```diff
@@ -6,6 +6,16 @@ namespace Common
 {
     public static class CenterOfMassTools
     {
+        public static float NormalizeToOne(float norm)
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Common/CenterOfMass.cs:9]: Mathf.Sign

File: JamGame/Assets/Scripts/TileBuilder/Command.cs (hunk 4 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/TileBuilder/Command.cs]: your new approach just replaces ref with wrapper class

**Review by mertwole — CHANGES_REQUESTED** [review]

File: JamGame/Assets/Scripts/Level/Sheduler.cs (hunk 2 above)
Comment (Kiuh) [reply JamGame/Assets/Scripts/Level/Sheduler.cs:21]: See Executor UnityEvent for understanding this comment

File: JamGame/Assets/Scripts/TileBuilder/Command.cs (hunk 4 above)
Comment (Kiuh) [reply JamGame/Assets/Scripts/TileBuilder/Command.cs]: Don't be drunk, its not final commit

And selected_tile used in all commands except one

Comment (mertwole) [reply JamGame/Assets/Scripts/TileBuilder/Command.cs]: https://github.com/mertwole/miniature-happiness/blob/931941b4ddd7a53cad063c4cb2154625b3966c1e/JamGame/Assets/Scripts/TileBuilder/Controller/Controller.cs#L16C9-L16C9

File: JamGame/Assets/Scripts/AnimatorsSwitcher/Model.cs
Comment (Kiuh) [reply JamGame/Assets/Scripts/AnimatorsSwitcher/Model.cs:1]: image
Comment this please

File: JamGame/Assets/Scripts/Level/Sheduler.cs (hunk 2 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/Level/Sheduler.cs:21]: line 21, what's Realese?

File: JamGame/Assets/Scripts/TileBuilder/Command.cs (hunk 4 above)
Comment (Kiuh) [reply JamGame/Assets/Scripts/TileBuilder/Command.cs]: Well, what don't you like?
It`s cover used not in all commands

File: JamGame/Assets/Scripts/AnimatorsSwitcher/Model.cs
Comment (mertwole) [reply JamGame/Assets/Scripts/AnimatorsSwitcher/Model.cs:1]: Boss will have view, when AnimatorsSwitcher will not

Comment (mertwole) [reply JamGame/Assets/Scripts/AnimatorsSwitcher/Model.cs:1]: Some classes called `Model` to distinguish them from `View`s and `Controller`s

File: JamGame/Assets/Scripts/Level/Sheduler.cs (hunk 2 above)
Comment (Kiuh) [reply JamGame/Assets/Scripts/Level/Sheduler.cs:21]: Relese

File: JamGame/Assets/Scripts/AnimatorsSwitcher/Model.cs
Comment (Kiuh) [reply JamGame/Assets/Scripts/AnimatorsSwitcher/Model.cs:1]: Oh, Okay

File: JamGame/Assets/Scripts/TileBuilder/Command.cs (hunk 4 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/TileBuilder/Command.cs]: It's more about assumption that your API makes (that almost all operations are performed on selected tile). In my opinion it shouldn't be like this

Comment (mertwole) [reply JamGame/Assets/Scripts/TileBuilder/Command.cs]: Actually, you don't really need selected tile at all, which will allow you to implement stateless API

File: JamGame/Assets/Scripts/Level/Sheduler.cs (hunk 2 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/Level/Sheduler.cs:21]: you shure?

File: JamGame/Assets/Scripts/TileBuilder/Command.cs [hunk 7]
```diff
@@ -33,11 +33,12 @@ SelectedTileCover selected_tile_cover
             selectedTileCover = selected_tile_cover;
         }
 
-        public AddTileToScene(TileUnionImpl tile_prefab)
+        public AddTileToScene(TileUnionImpl tile_prefab, SelectedTileWrapper selected_tile_cover)
```
Comment (mertwole) [inline JamGame/Assets/Scripts/TileBuilder/Command.cs:36]: `selected_tile_wrapper`

Comment (mertwole) [reply JamGame/Assets/Scripts/TileBuilder/Command.cs:36]: and everywhere in this file

File: JamGame/Assets/Scripts/TileUnion/Tile/TileImpl.cs [hunk 8]
```diff
@@ -72,6 +76,28 @@ private void OnValidate()
...
+            Handles.Label(
+                transform.position + transform.TransformDirection(new Vector3(1, 0, 0) * 2),
+                "Up"
+            );
+            Handles.Label(
+                transform.position + transform.TransformDirection(new Vector3(-1, 0, 0) * 2),
+                "Down"
+            );
+            */
+        }
```
Comment (mertwole) [inline JamGame/Assets/Scripts/TileUnion/Tile/TileImpl.cs:99]: If this is for debug purpose, add some flag to enable/disable gizmo draw

**Review by mertwole — CHANGES_REQUESTED** [review]

File: JamGame/Assets/Scripts/Level/Sheduler.cs (hunk 2 above)
Comment (Kiuh) [reply JamGame/Assets/Scripts/Level/Sheduler.cs]: Activation

**Pull request merged**